src/replication/log_consumer.hpp

```diff
@@ -163,6 +163,17 @@ namespace cubreplication
       subtran_applier &get_subtran_applier ();
   };
 
+  //
+  // dispatch_consumer is the common interface used by dispatch to control group creation.
+  //
+  class dispatch_consumer
```

Comment:
1. Does this class have any meaning outside slave group control manager? The virtual interface and multiple inheritance of slave GCM makes it undesirable unless there are strong arguments to keep the class separate from slave GCM.

2. In case we do need the separate class, its name conflicts with the dispatcher of tasks. Is there a better name?

Reply:
dispatch_consumer is a interface implemented by slave manager, used by consumer to notify slave manager. Please, specify a better name. Also whether you think is better to move it in other file.

Reply:
The instantiation of dispatch_consumer  is :
`m_p_dispatch_consumer = cubtx::slave_group_complete_manager::get_instance ();`

If it would be something like :
`m_p_dispatch_consumer = cubtx::complete_manager::get_instance ();`

The interface would be required.
With current code, the methods from this interface can be used only with slave GCM.

Reply:
Again, with the interface, the consumer can calls only methods that he needs. The consumer can't call other public methods from slave manager, to not corrupt the system (the consumer can't call do_complete for instance). We can move the interface in other file, rename it and I will do these changes.

Reply:
I understand. Still virtual interface and multiple inheritance remain; these can be avoided by using composition instead of inheritance to limit the access to only a subset of public functions:
```c
  slave_gcm->get_component_with_subset_interface ();
```
If it is not easy to do it in place, open an issue to consider it for later.

A general note here, we need to discuss and document this type of model in an updated code guideline.

Reply:
One related discussion was Multiple Inheritance vs. Composition :
http://jira.cubrid.org/browse/CBRD-23218